src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetPowerShellUsageCollector.cs

```diff
@@ -0,0 +1,371 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.VisualStudio.Telemetry.Powershell;
+
+namespace NuGet.VisualStudio.Telemetry
+{
+    public sealed class NuGetPowerShellUsageCollector : IDisposable
+    {
+        // PMC, PMUI powershell telemetry consts
+        public const string NuGetPMCExecuteCommandCount = nameof(NuGetPMCExecuteCommandCount);
+        public const string NuGetPMUIExecuteCommandCount = nameof(NuGetPMUIExecuteCommandCount);
+        public const string NuGetCommandUsed = nameof(NuGetCommandUsed);
+        public const string InitPs1LoadPMUI = nameof(InitPs1LoadPMUI);
+        public const string InitPs1LoadPMC = nameof(InitPs1LoadPMC);
+        public const string InitPs1LoadedFromPMCFirst = nameof(InitPs1LoadedFromPMCFirst);
+        public const string LoadedFromPMUI = nameof(LoadedFromPMUI);
+        public const string FirstTimeLoadedFromPMUI = nameof(FirstTimeLoadedFromPMUI);
+        public const string LoadedFromPMC = nameof(LoadedFromPMC);
+        public const string FirstTimeLoadedFromPMC = nameof(FirstTimeLoadedFromPMC);
+        public const string SolutionLoaded = nameof(SolutionLoaded);
+        public const string NuGetPowerShellLoaded = nameof(NuGetPowerShellLoaded);
+
+        // PMC UI Console Container telemetry consts
+        public const string NuGetPMCWindowLoadCount = nameof(NuGetPMCWindowLoadCount);
+        public const string ReOpenAtStart = nameof(ReOpenAtStart);
+
+        // Const name for emitting when VS solution close or VS instance close.
+        public const string NuGetPowershellPrefix = "NuGetPowershellPrefix."; // Using prefix prevent accidental same name property collission from different type telemetry.
```

Comment:
I would try to reduce how many times 'nuget' appears in event and property names.
```
        public const string PackageCommands = "PackageCommands."; // Using prefix prevent accidental same name property collission from different type telemetry.

```

Reply:
The word "prefix" in the property names doesn't add anything to the name either.

I'm not sure if `PackageCommands.` is the suggestion, or just a generic example, but this telemetry is not about packages, or really about commnads. It's more about NuGet's powershell host usage, so I think `vs.nuget.powershellhost.` would be a good prefix.

Reply:
From [this comment thread](https://github.com/NuGet/NuGet.Client/pull/3755#discussion_r562093185), Fernando suggested keeping the generic "vs.nuget.solutionclose" and "vs.nuget.instanceclose" event names, so we can reuse it for other data in the future where we want to report at solution/instance close.